File: src/main/java/net/ndrei/teslacorelib/inventory/IFilteredFluidTank.java
package net.ndrei.teslacorelib.inventory;

import net.minecraftforge.fluids.FluidStack;
import net.minecraftforge.fluids.IFluidTank;

import javax.annotation.Nonnull;

/**
 * Created by CF on 2016-12-29.
 */
public interface IFilteredFluidTank extends IFluidTank {
    boolean acceptsFluid(@Nonnull FluidStack fluid);
    boolean canDrain();
    boolean canFill();
}

File: src/main/java/net/ndrei/teslacorelib/inventory/IFilteredFluidTank.kt
package net.ndrei.teslacorelib.inventory

import net.minecraftforge.fluids.FluidStack
import net.minecraftforge.fluids.IFluidTank

/**
 * Created by CF on 2017-06-28.
 */
interface IFilteredFluidTank : IFluidTank {
    fun acceptsFluid(fluid: FluidStack): Boolean
    fun canDrain(): Boolean
    fun canFill(): Boolean
}
